Public/Src/FrontEnd/Script/Ast/Values/ObjectLiterals/ArrayLiteral.cs

```diff
@@ -191,7 +191,7 @@ public override bool TryProject(Context context, SymbolAtom name, ModuleLiteral
                 return true;
             }
 
-            var resolvedMember = context.PredefinedTypes.AmbientArray.ResolveMember(this, name);
+            var resolvedMember = ((ModuleRegistry)context.FrontEndHost.ModuleRegistry).PredefinedTypes.AmbientArray.ResolveMember(this, name);

```

Comment:
>(ModuleRegistry) [](start = 34, length = 16)

Maybe add TODOs in all these places were abstractions are broken

Reply:
I was planning to search for all Casts when cleaning it up. Should be pretty clearn

Reply:
Sure

---
In reply to: [276370271](https://github.com/Microsoft/BuildXL/pull/155#discussion_r276370271) [](ancestors = 276370271)